src/DropdownMenu.js

```diff
@@ -1,43 +1,155 @@
-import React, { cloneElement } from 'react';
+import React from 'react';
+import domUtils from './utils/domUtils';
+import EventListener from './utils/EventListener';
+import keycode from 'keycode';
 import classNames from 'classnames';
-
 import createChainedFunction from './utils/createChainedFunction';
-import ValidComponentChildren from './utils/ValidComponentChildren';
 
-const DropdownMenu = React.createClass({
-  propTypes: {
-    pullRight: React.PropTypes.bool,
-    onSelect: React.PropTypes.func
-  },
+class DropdownMenu extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.focusNext = this.focusNext.bind(this);
+    this.focusPrevious = this.focusPrevious.bind(this);
+    this.getFocusableMenuItems = this.getFocusableMenuItems.bind(this);
+    this.getItemsAndActiveIndex = this.getItemsAndActiveIndex.bind(this);
+
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+
+    this.bindRootCloseHandler = this.bindRootCloseHandler.bind(this);
+    this.unbindRootCloseHandler = this.unbindRootCloseHandler.bind(this);
+    this.handleDocumentClick = this.handleDocumentClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.componentDidUpdate();
+  }
+
+  componentDidUpdate() {
+    if (this.props.open) {
+      this.bindRootCloseHandler();
+    } else {
+      this.unbindRootCloseHandler();
+    }
+  }
+
+  bindRootCloseHandler() {
+    let ownerDocument = domUtils.ownerDocument(this);
+
+    this._onDocumentClickListener =
+      EventListener.listen(ownerDocument, 'click', this.handleDocumentClick);
+  }
+
+  unbindRootCloseHandler() {
+    if (this._onDocumentClickListener) {
+      this._onDocumentClickListener.remove();
+    }
+  }
+
+  componentWillUnmount() {
+    this.unbindRootCloseHandler();
+  }
+
+  handleDocumentClick(event) {
+    let inTree = domUtils.isNodeInTree(event.target, React.findDOMNode(this));
```

Comment:
I think this is probably now a duplicate of `domUtils.contains` which was added recently-ish